lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1388,6 +1431,11 @@ func (d *zfs) GetVolumeUsage(vol Volume) (int64, error) {
 // SetVolumeQuota sets the quota/reservation on the volume.
 // Does nothing if supplied with an empty/zero size for block volumes.
 func (d *zfs) SetVolumeQuota(vol Volume, size string, allowUnsafeResize bool, op *operations.Operation) error {
+	// Quota does not apply to zfs block volumes.
+	if d.Info().BlockBacking {
```

Comment:
Because the volume.zfs.block_mode` setting will be copied into each volume's DB config entry as `zfs.block_mode`, and at any given time there can be volumes that are in a different mode to the pool's current setting, when operating on a specific existing volume all decisions around block mode need to use the volume's config not the pool's setting.

Also given that currently the ZFS driver always has `BlockBacking=false` I'm not sure why this logic is here.

Furthermore, I'm not seeing how we would support resizing the block filesystem volumes, as `SetVolumeQuota` is an exported function and part of the driver interface so does need to be able to handle block backed filesystem volumes. It will need to use similar logic to the LVM driver, in that it needs to detect whether the filesystem being used supports shrinking etc.